File: views/termopanel.go
package views

import (
	"html/template"
	"net/http"

	mobiledetect "github.com/Shaked/gomobiledetect"
	"github.com/meehalkoff/loggi"
	"gitlab.com/mvalkov/katzman/lib"
)

func TermopanelHandler(w http.ResponseWriter, r *http.Request) {

	tmpl := template.New("termopanel")

	detect := mobiledetect.NewMobileDetect(r, nil)
	if detect.IsMobile() {
		tmpl.ParseFiles(
			"mobile/templates/termopanel.html",
			"mobile/templates/header.html",
			"mobile/templates/footer.html",
			"mobile/templates/top-panel.html",
			"mobile/templates/calc.html",
			"mobile/templates/call.html",
			"mobile/templates/about.html",
			"mobile/templates/firstPromoBlock.html",
			"mobile/templates/creditPromoBlock.html",
			"mobile/templates/lastPromoBlock.html",
		)
	} else {
		tmpl.ParseFiles(
			"templates/header.html",
			"templates/footer.html",
			"templates/top-panel/top-panel.html",
			"templates/top-panel/about.html",
			"templates/top-panel/calc.html",
			"templates/top-panel/call.html",
			"templates/top-panel/main-menu.html",
			"templates/control-panel.html",
			"templates/firstPromoBlock.html",
			"templates/termopanel/index.html",
			"templates/termopanel/summary.html",
			"templates/termopanel/palitra.html",
			"templates/termopanel/kachestva.html",
			"templates/termopanel/schema.html",
			"templates/termopanel/otkos.html",
			"templates/howWeWork.html",
			"templates/termopanel/portfolio.html",
			"templates/shared/video.html",
			"templates/termopanel/promo.html",
			"templates/lastPromoBlock.html",
			"templates/creditPromoBlock.html",
		)
	}
	data := lib.FirstPromoBlock{
		MaterialName: `Термопанели`,
		ID:           "termoPrice",
		PictureURL:   "/assets/img/promo/termopanel.jpg",
		UnitName:     "кв.м.",
		YoutubeID:    "otvAAro2qIU",
	}
	if err := tmpl.Execute(w, data); err != nil {
		loggi.Fatal(err)
	}
}

File: assets/js/termopanel.js
var nextArrow = '<div class="control-next-arrow">▶▶</div>';
var prevArrow = '<div class="control-prev-arrow">◀◀</div>';
var maxSlidePosition = 10;

var control = {
    0: {
        id: "first-promo",
        title: "Термопанели: предложение",
        prev: "",
        next: "Виды покрытий" + nextArrow,
    },
    1: {
        id: "summary",
        title: "Термопанели: обзор",
        prev: prevArrow + "Предложение",
        next: "Палитра цветов" + nextArrow,
    },
    2: {
        id: "palitra",
        title: "Термопанели: палитра цветов",
        prev: prevArrow + "Обзор",
        next: "Качества" + nextArrow,
    },
    3: {
        id: "kachestva",
        title: "Термопанели: качества",
        prev: prevArrow + "Палитра цветов",
        next: "Технология монтажа" + nextArrow,
    },
    4: {
        id: "schema",
        title: "Термопанели: монтаж",
        prev: prevArrow + "Качества",
        next: "Устройство откосов" + nextArrow,
    },
    5: {
        id: "otkos",
        title: "Термопанели: откосы",
        prev: prevArrow + "Технология монтажа",
        next: "Как мы работаем" + nextArrow,
    },
    6: {
        id: "how-we-work",
        title: "Термопанели: как мы работаем",
        prev: prevArrow + "Откосы",
        next: "Портфолио" + nextArrow,
    },
    7: {
        id: "portfolio",
        title: "Термопанели: портфолио",
        prev: prevArrow + "Устройство откосов",
        next: "Видео обзор" + nextArrow,
    },
    8: {
        id: "video",
        title: "Термопанели: видео обзор",
        prev: prevArrow + "Портфолио",
        next: "Предложение" + nextArrow,
    },
    9: {
        id: "credit-promo",
        title: "Термопанели: рассрочка",
        prev: prevArrow + "Видео",
        next: "Предложение" + nextArrow,
    },
    10: {
        id: "last-promo",
        title: "Термопанели: звоните!",
        prev: prevArrow + "Рассрочка",
        next: "",
    },
}

var maxPortfolioPosition = 5;

var portfolio = {
    0: 'url(/assets/img/termopanel/portfolio/0.jpg)',
    1: 'url(/assets/img/termopanel/portfolio/1.jpg)',
    2: 'url(/assets/img/termopanel/portfolio/2.jpg)',
    3: 'url(/assets/img/termopanel/portfolio/3.jpg)',
    4: 'url(/assets/img/termopanel/portfolio/4.jpg)',
    5: 'url(/assets/img/termopanel/portfolio/5.jpg)',
    // 6: 'url(/assets/img/termopanel/portfolio/6.jpg)',
    // 7: 'url(/assets/img/termopanel/portfolio/7.jpg)',
}